lib/util/binary-diff.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var fs = require('fs');
+var readChunk = require('read-chunk');
+var istextorbinary = require('istextorbinary');
+var dateFormat = require('dateformat');
+var prettyBytes = require('pretty-bytes');
+var Table = require('cli-table');
+
+module.exports = {
+  isBinary: function (existingFilePath, newFileContents) {
+    var existingHeader = readChunk.sync(existingFilePath, 0, 512);
+    return istextorbinary.isBinarySync(undefined, existingHeader) || istextorbinary.isBinarySync(undefined, newFileContents);
+  },
+
+  diff: function (existingFilePath, newFileContents) {
+    var existingStat = fs.statSync(existingFilePath);
+
+    var table = new Table({
+      head: ['', 'Existing', 'Replacement', 'Diff']
+    });
+
+    var sizeDiff;
+    if (existingStat.size > newFileContents.length) {
+      sizeDiff = '-';
+    } else {
+      sizeDiff = '+';
+    }
+    sizeDiff += prettyBytes(Math.abs(existingStat.size - newFileContents.length));
+
+    table.push(
+      ['Size',
+        prettyBytes(existingStat.size),
+        prettyBytes(newFileContents.length),
+        sizeDiff],
+      ['Last modified',
+        dateFormat(existingStat.mtime),
+        'N/A',
```

Comment:
I'm not following. @SBoudrias 

But isn't it just about passing `file` instead of `file.contents` here https://github.com/yeoman/generator/pull/737/files#diff-0765c740f9a715d47c62655d99f00ee6R171 ?